src/main/habit-modal-dialogue/habit-modal-dialogue.component.ts

```diff
@@ -0,0 +1,211 @@
+  }
+
+  updateGoalFrequency(selectedFrequency: string) {
+    this.goalFrequency = selectedFrequency;
+  }
+
+  getDayValue(): string {
+    const allDays = Object.values(daysOfWeek);
+    if (this.checkRepeat == AppConstants.days) {
+      if (this.areArraysEqual(this.days, allDays)) {
+        this.repeatValue = AppConstants.Daily;
+        return AppConstants.Daily;
+      } else {
+        this.repeatValue = this.days.join(' ');
+        const selectedDays = this.days.map(day => day.substring(0, 3)); // Abbreviate to three letters
+        return selectedDays.join(',');
+      }
+    } else if (this.checkRepeat == AppConstants.interval) {
+      this.repeatValue = this.intervalPerDays;
+      return this.intervalPerDays;
+    } else {
+      this.repeatValue = this.selectedDates.map(date => date.toLocaleDateString('en-US')).join(',');
+      return this.months;
+    }
+  }
+
+  areArraysEqual(firstArray: string[], secondArray: string[]): boolean {
+    if (firstArray.length !== secondArray.length) {
+      return false;
+    }
+
+    const sortedArr1 = firstArray.slice().sort();
+    const sortedArr2 = secondArray.slice().sort();
+
+    return sortedArr1.every((value, index) => value === sortedArr2[index]);
+  }
+
+  updateFrequencyPeriod(selectedFrequency: string) {
+    this.frequencyPerPeriod = selectedFrequency;
+  }
+
+  updateTimeOfDay(selectedTime: TimeOfDay) {
+    const index = this.timeOfDay.indexOf(selectedTime);
+
+    if (index !== -1) {
+      // If the time is already in the array, check if there are more than one item
+      if (this.timeOfDay.length > 1) {
+        // If there is more than one item, remove it
+        this.timeOfDay.splice(index, 1);
+      }
+      // If there is only one item, do nothing (prevent removal)
+    } else {
+      // If the time is not in the array, add it
+      this.timeOfDay.push(selectedTime);
+    }
+  }
+
+  isTimeOfDaySelected(time: TimeOfDay): boolean {
+    return this.timeOfDay.includes(time);
+  }
+
+  getTimeOfDayDisplayValue(timeOfDay: TimeOfDay[]): string {
+    const allTimesIncluded = timeOfDay.length === 3 && timeOfDay.includes(TimeOfDay.Morning) && timeOfDay.includes(TimeOfDay.Afternoon) && timeOfDay.includes(TimeOfDay.Evening);
+
+    if (allTimesIncluded) {
+      return TimeOfDay.Anytime;
+    } else {
+      return timeOfDay.join(', ');
+    }
+  }
+
+  getCalenderDisplayValue(date: Date | undefined): string {
+    return this.displayService.getCalenderDialogueDisplayValue(date);
+  }
+
+  onCalendarClick(event: Event) {
+    const clickedInside = this.calenderDialogue.nativeElement?.contains(event.target);
+    if (!clickedInside) {
+      event.stopPropagation();
+    }
+  }
+
+  toggleCalendar(): void {
+    this.showCalendar = !this.showCalendar;
+  }
+
+  onDateSelect(date: Date) {
+    // Check if the date is already present in the array
+    const isDateSelected = this.selectedDates.some(d => d.getDate() === date.getDate() && d.getMonth() === date.getMonth());
```

Comment:
try to use more descriptive variable names instead of single letter like above "d"

Reply:
updated